mpas_analysis/ocean/streamfunction_moc.py

```diff
@@ -338,6 +337,327 @@ def _load_mesh(self):  # {{{
+                             'either timeSeriesStatsMonthlyOutput or '
+                             'mocStreamfunctionOutput streams')
+
+        binBoundaryMocStreamfunction = np.rad2deg(binBoundaryMocStreamfunction)
+        dLat = binBoundaryMocStreamfunction - 26.5
+        indlat26 = np.where(dLat == np.amin(np.abs(dLat)))
+
+        inputFilesTseries = sorted(self.historyStreams.readpath(
+                streamName, startDate=self.startDateTseries,
+                endDate=self.endDateTseries, calendar=self.calendar))
+
+        years, months = get_files_year_month(inputFilesTseries,
+                                             self.historyStreams,
+                                             'timeSeriesStatsMonthlyOutput')
+
+        mocRegion = np.zeros(len(inputFilesTseries))
+        times = np.zeros(len(inputFilesTseries))
+        computed = np.zeros(len(inputFilesTseries), bool)
+
+        continueOutput = os.path.exists(outputFileTseries)
+        if continueOutput:
+            self.logger.info('   Read in previously computed MOC time series')
+            with open_mpas_dataset(fileName=outputFileTseries,
+                                   calendar=self.calendar,
+                                   timeVariableNames=None,
+                                   variableList=['mocAtlantic26'],
+                                   startDate=self.startDateTseries,
+                                   endDate=self.endDateTseries) as dsMOCIn:
+
+                dsMOCIn.load()
+
+                # first, copy all computed data
+                for inIndex in range(dsMOCIn.dims['Time']):
+
+                    mask = np.logical_and(
+                            dsMOCIn.year[inIndex].values == years,
+                            dsMOCIn.month[inIndex].values == months)
+
+                    outIndex = np.where(mask)[0][0]
+
+                    mocRegion[outIndex] = dsMOCIn.mocAtlantic26[inIndex]
+                    times[outIndex] = dsMOCIn.Time[inIndex]
+                    computed[outIndex] = True
+
+                if np.all(computed):
+                    # no need to waste time writing out the data set again
+                    return dsMOCIn
+
+        for timeIndex, fileName in enumerate(inputFilesTseries):
+            if computed[timeIndex]:
+                continue
+
+            dsLocal = open_mpas_dataset(
+                fileName=fileName,
+                calendar=self.calendar,
+                variableList=self.variableList,
+                startDate=self.startDateTseries,
+                endDate=self.endDateTseries)
+            dsLocal = dsLocal.isel(Time=0)
+            time = dsLocal.Time.values
+            times[timeIndex] = time
+            date = days_to_datetime(time, calendar=self.calendar)
+
+            self.logger.info('     date: {:04d}-{:02d}'.format(date.year,
+                                                               date.month))
+
+            # hard-wire region=0 (Atlantic) for now
```

Comment:
This whole region handling will change substantially once we use the regional capability fully.

Reply:
Yes, see #246